### PR TITLE
Update Skill Template for new Manifest / SDK GA Changes

### PR DESCRIPTION
**Pull request by darrenj**
This PR incorporates all of the manual changes tested against Power Virtual Agents enabling a Skill created using this template to work OOB with P-VA and also adds action support.

- New manifest added to wwwroot/manifest folder
- Removed old manifestTemplate.json (therefore a new skill will require the new updates to botskills)
- Removed unnecessary adapter code
- Add EndOfConversation support
- Added SampleAction and wiring code
- Updated sdk libs to 4.7.1

**Followup areas** 
- Update Manifest schema to v2.1 and add dispatchModels element
- Incorporate MainDialog changes to better centralise the endofconverastion and remove turnstate code

### CODE REVIEW

**Review by lauren-mills — COMMENTED** [review]
Were there any changes that need to be made in the test project template?

File: templates/csharp/Skill/Skill/Bots/DefaultActivityHandler.cs [hunk 1]
```diff
@@ -30,6 +30,16 @@ public DefaultActivityHandler(IServiceProvider serviceProvider, T dialog)
 
         public override async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default)
         {
+            var activity = turnContext.Activity;
```
Comment (lzc850612) [inline templates/csharp/Skill/Skill/Bots/DefaultActivityHandler.cs]: This is done in the latest SkillMiddleware in Microsoft.Bot.Solutions so we don't have to do it here

File: templates/csharp/Skill/Skill/Controllers/BotController.cs [hunk 2]
```diff
@@ -7,20 +7,28 @@
 using Microsoft.Bot.Builder.Solutions;
 using Microsoft.Bot.Builder.Solutions.Skills;
 using Microsoft.Bot.Builder.Solutions.Skills.Auth;
+using System.Threading.Tasks;
 
 namespace $safeprojectname$.Controllers
 {
+     [Route("api/messages")]
```
Comment (lzc850612) [inline templates/csharp/Skill/Skill/Controllers/BotController.cs]: there's an extra space :)

File: templates/csharp/Skill/Skill/Dialogs/MainDialog.cs [hunk 3]
```diff
@@ -161,25 +163,40 @@ protected override async Task OnMessageActivityAsync(DialogContext innerDc, Canc
         }
 
         // Runs when a new event activity comes in.
-        protected override async Task OnEventActivityAsync(DialogContext innerDc, CancellationToken cancellationToken = default)
+        protected override async Task OnEventActivityAsync(DialogContext dc, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (lzc850612) [inline templates/csharp/Skill/Skill/Dialogs/MainDialog.cs]: i think the default annotation is the new one. we don't need the (CancellationToken) part.

File: templates/csharp/Skill/Skill/Dialogs/MainDialog.cs [hunk 4]
```diff
@@ -192,31 +209,19 @@ protected override async Task OnUnhandledActivityTypeAsync(DialogContext innerDc
         // Runs when the dialog stack completes.
         protected override async Task OnDialogCompleteAsync(DialogContext outerDc, object result, CancellationToken cancellationToken)
         {
-            if (outerDc.Context.Adapter is IRemoteUserTokenProvider || outerDc.Context.Activity.ChannelId != Channels.Msteams)
+            // Retrieve the prior dialogs result if provided to return on the Skill EndOfConversation event.
```
Comment (lzc850612) [inline templates/csharp/Skill/Skill/Dialogs/MainDialog.cs:212]: so are we still going to put in the new MainDialog by Lauren?

File: templates/csharp/Skill/Skill/wwwroot/manifest/manifest.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
+{
+  "$schema": "https://schemas.botframework.com/schemas/skills/skill-manifest-2.0.0.json",
```
Comment (lzc850612) [inline templates/csharp/Skill/Skill/wwwroot/manifest/manifest.json:2]: should the version be 2.0.0. or 2.1.0?

**Review by lzc850612 — CHANGES_REQUESTED** [review]

**Comment by darrenj** [comment]
> Were there any changes that need to be made in the test project template?

Yes :-)  Now done!

File: templates/csharp/Skill/Skill/Bots/DefaultActivityHandler.cs (hunk 1 above)
Comment (darrenj) [reply templates/csharp/Skill/Skill/Bots/DefaultActivityHandler.cs]: Removed

File: templates/csharp/Skill/Skill/Controllers/BotController.cs (hunk 2 above)
Comment (darrenj) [reply templates/csharp/Skill/Skill/Controllers/BotController.cs]: removed

File: templates/csharp/Skill/Skill/Dialogs/MainDialog.cs (hunk 3 above)
Comment (darrenj) [reply templates/csharp/Skill/Skill/Dialogs/MainDialog.cs]: good spot

File: templates/csharp/Skill/Skill/Dialogs/MainDialog.cs (hunk 4 above)
Comment (darrenj) [reply templates/csharp/Skill/Skill/Dialogs/MainDialog.cs:212]: Yes that will follow next

File: templates/csharp/Skill/Skill/wwwroot/manifest/manifest.json (hunk 5 above)
Comment (darrenj) [reply templates/csharp/Skill/Skill/wwwroot/manifest/manifest.json:2]: 2.0.0 for now, 2.1.0 later this week / next week

**Review by lzc850612 — APPROVED** [review]

**Review by lauren-mills — APPROVED** [review]

**Pull request merged**